### ISSUE TITLE
[FEATURE] java-client should support FunctionScore with MatchTermQuery

### ISSUE THREAD

**Issue by m-hofmann** [opensearch-project/opensearch-java#655]
We are trying to use java-client 2.6.0 to build a function score query boosting specific hits on fields. Our query JSON looks like this:

```json
{
  "from": 0,
  "size": 20,
  "query": {
    "function_score": {
      "query": {
        "bool": {
          "should": [
            {
              "match": {
                "some_field": "1234"
              }
            }
          ],
          "minimum_should_match": 1
        }
      },
      "boost_mode": "multiply",
      "score_mode": "sum",
      "functions": [
        {
          "filter": {
            "match": {
              "some_field": "1234"
            }
          },
          "weight": 2
        }
      ]
    }
  }
}
```

unfortunately the java-client seems to support only specific function "kinds" for the functions block, like `random_score` or `script_score`. When we try to instantiate a query using the Java DSL that looks like the above snippet: 

```
FunctionScore functionScore = new FunctionScore.Builder()
   .new ContainerBuilder()
   .filter(query)
   .weight(10.0)
   .build();
```

an exception `org.opensearch.client.util.MissingRequiredPropertyException: Missing required property 'Builder.<variant kind>` will be thrown. 

### What solution would you like?

Being able to build a function object combining a (match)query with a weight. 

### What alternatives have you considered?

We tried to disable API warnings using `ApiTypeHelper.DANGEROUS_disableRequiredPropertiesCheck(true)` but then serialization of the request in the search client fails.

Therefore we're switching back to the REST highlevel client for now.

**Comment by jmontgomery-wbd** [comment]
any updates on this? this is a pretty big miss....

**Comment by jmontgomery-wbd** [comment]
@dblock  it seems like there is a lot of red tape around contributing.  I just pulled and made an update and it seems to have fixed it: 
<img width="1408" alt="Screen Shot 2023-11-03 at 4 13 13 PM" src="https://github.com/opensearch-project/opensearch-java/assets/72515781/e1c8080a-91ad-4c10-92db-de44b34a0a85">

how can we properly test and merge that?

**Comment by dblock** [comment]
@jmontgomery-wbd No tape. Just make a pull request with a passing test that wasn't passing before.

**Comment by VachaShah** [comment]
This feature was added in PR #880 and will be released in the upcoming version. Currently the change is available to consume in [2.10.0-SNAPSHOT](https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/client/opensearch-java/2.10.0-SNAPSHOT/).
